dear_petition/petition/models.py

```diff
@@ -220,11 +230,22 @@ def age(self):
         today = timezone.now().date()
         return today.year - dob.year - ((today.month, today.day) < (dob.month, dob.day))
 
+    @property
+    def automatic_delete_date(self):
+        """Date when this batch will be automatically deleted by the cleanup task"""
+        return self.date_uploaded + timedelta(
+            hours=settings.CIPRS_RECORD_LIFETIME_HOURS
+        )
+
+
+def get_batch_file_upload_path(instance, filename):
+    return "batch/%s/%s" % (instance.batch.id, filename)
+
 
 class BatchFile(models.Model):
     batch = models.ForeignKey(Batch, related_name="files", on_delete=models.CASCADE)
     date_uploaded = models.DateTimeField(auto_now_add=True)
-    file = models.FileField(upload_to="ciprs/")
+    file = models.FileField(upload_to=get_batch_file_upload_path)
```

Comment:
How is `filename` getting passed here? I feel like this is a django thing i'm not understanding

Reply:
It's passed to the function:

https://github.com/deardurham/dear-petition/blob/029e1533f9c626ffc07237c275062c81381948d3/dear_petition/petition/models.py#L241-L242